pkg/ocm/clusters.go

```diff
@@ -558,3 +558,62 @@ func (c *Client) createClusterSpec(config Spec, awsClient aws.Client) (*cmv1.Clu
 
 	return clusterSpec, nil
 }
+
+func (c *Client) HibernateCluster(clusterKey string, creatorARN string) error {
+	reporter := rprtr.CreateReporterOrExit()
+
+	err := c.IsHibernateCapabilityEnabled()
+	if err != nil {
+		return err
+	}
+
+	// Get the cluster to check the state
+	cluster, err := c.GetCluster(clusterKey, creatorARN)
+	if err != nil {
+		reporter.Errorf("Failed to get cluster '%s': %v", clusterKey, err)
+		os.Exit(1)
+	}
+
+	if cluster.State() != cmv1.ClusterStateReady {
```

Comment:
I think we should check this in the command. By the time you get to this function we should let the API be the one returning the error.